src/runtime/Converter.cs

```diff
@@ -9,250 +9,253 @@
+    /// * Managed converters <see cref="ToManaged"/>, <see cref="ToManagedValue"/>,
+    ///   <see cref="ToManagedExplicit"/>, <see cref="ToPrimitive"/>,
+    ///   <see cref="ToPyObjectSubclass"/>, <see cref="ToArray"/> and <see cref="ToInt32"/>
+    /// * Functions to determine whether to use user converters; <see cref="DecodableByUser"/>
+    ///   and <see cref="EncodableByUser"/>
     /// </summary>
     [SuppressUnmanagedCodeSecurity]
-    internal class Converter
+    internal static class Converter
     {
-        private Converter()
-        {
-        }
-
-        private static readonly Type objectType;
-        private static readonly Type stringType;
-        private static readonly Type singleType;
-        private static readonly Type doubleType;
-        private static readonly Type int16Type;
-        private static readonly Type int32Type;
-        private static readonly Type int64Type;
-        private static readonly Type boolType;
-        private static readonly Type typeType;
-
-        static Converter()
-        {
-            objectType = typeof(Object);
-            stringType = typeof(String);
-            int16Type = typeof(Int16);
-            int32Type = typeof(Int32);
-            int64Type = typeof(Int64);
-            singleType = typeof(Single);
-            doubleType = typeof(Double);
-            boolType = typeof(Boolean);
-            typeType = typeof(Type);
-        }
-
+        internal delegate bool TryConvertFromPythonDelegate(
+            BorrowedReference pyObj, out object? result
+        );
+
+        #region static members
+        private static readonly Type OBJECT_TYPE = typeof(Object);
+        private static readonly Type STRING_TYPE = typeof(String);
+        private static readonly Type SINGLE_TYPE = typeof(Single);
+        private static readonly Type DOUBLE_TYPE = typeof(Double);
+        private static readonly Type INT16_TYPE = typeof(Int16);
+        private static readonly Type INT32_TYPE = typeof(Int32);
+        private static readonly Type INT64_TYPE = typeof(Int64);
+        private static readonly Type BOOL_TYPE = typeof(Boolean);
+        private static readonly Type CLR_TYPE = typeof(Type);
+
+        private static readonly Func<object, bool> IS_TRANSPARENT_PROXY = GetIsTransparentProxyFunc();
+        #endregion
 
         /// <summary>
         /// Given a builtin Python type, return the corresponding CLR type.
         /// </summary>
+        /// <param name="op">The Python type</param>
+        /// <returns>The CLR type</returns>
         internal static Type? GetTypeByAlias(BorrowedReference op)
         {
-            if (op == Runtime.PyStringType)
-                return stringType;
-
-            if (op == Runtime.PyUnicodeType)
-                return stringType;
-
-            if (op == Runtime.PyLongType)
-                return int32Type;
-
-            if (op == Runtime.PyLongType)
```

Comment:
This block can never be reached so it's removed ; PyLongType maps to INT32_TYPE now